test/functionalTest/cases/0501_cors/allowed_headers.test

```diff
@@ -40,6 +40,42 @@ orionCurl --url /v2/entities -X OPTIONS
 echo
 echo
 
+echo "03. Allowed headers: GET DELETE OPTIONS"
```

Comment:
As far as I undertand, https://github.com/telefonicaid/fiware-orion/pull/3032/files#diff-c7a50c3b97af231671cfad8674374d21R844 is adding 10 new routes (i.e. 10 new URLs supporting OPTIONS).

However, only 6 new URL/steps have been included...

Maybe I'm missing something? :)

Reply:
We had discussed about this more than a few times, it is normal to get confused :)

Right now we have 12 different routes for /v2 which have 8 different verb sets. 

As an example, by having `optionsGetPostOnly` handler, we cover both `/v2/entities` and `/v2/subscriptions`. So that's why we have less handlers than the number of routes, similar to bad verb handlers.

Reply:
Ok, I think I see now your point...

Note functional tests should be designed from usage/API perspective, not implementation perspective. I mean, I agree with you that in this particular case and with the current implementation, with just the 8 operations you include in the test you can cover the 8 methods options handlers.

However, imagine that at some point and for some unknown reason at the present moment the implementation changes (but not the API). You would need to re-think the test in order to align with the new implementation. Thus, if the test is designed with API perspective from the very beginning, you would change the implementation without changing test... which at the end is the actual purpose of functional testing ;)

Moreover, in this case the modification in the .test is pretty simple. Just adding the 4 missing cases.

Reply:
It seems a similar situation occurs with cors_bad_verbs.test. It would be great (and easy :) to pass from 8 to 12 URLs also in that test.

Reply:
And now... I see your point :) 

You are absolutely right, thanks for pointing this out. I'll fix this in the next commit.

Reply:
Oh.. I just realized the name of this test file should have been allowed_methods. I will fixed that as well.

Reply:
Fixed in b9a8b2f